### ISSUE TITLE
Unable to configure ClaimMapping/ListClaimMapping for JWT/OIDC auth methods

### ISSUE THREAD

**Issue by pbusko** [terraform-providers/terraform-provider-consul#207]
Hello,

Due to fact that "config" attribute schema is TypeMap with simple string values it's impossible to provide nested map for ClaimMapping/ListClaimMapping required for JWT/OIDC auth methods

```go
"config": {
				Type:        schema.TypeMap,
				Required:    true,
				Description: "The raw configuration for this ACL auth method.",
				Elem: &schema.Schema{
					Type: schema.TypeString,
				},
			},
```

### Terraform Version
```
Terraform v0.12.26
terraform-consul-provider v2.8.0
```

### Affected Resource(s)
- consul_acl_auth_method

### Terraform Configuration Files
```hcl
resource "consul_acl_auth_method" "jwt" {
  name        = "jwt"
  type        = "jwt"
  description = "Auth method based on JWT tokens"

  config = {
    JWKSURL          = "https://somedomain.com/identity/oidc/.well-known/keys"
    JWTSupportedAlgs = "RS256"
    BoundIssuer      = "https://somedomain.com"
    ClaimMappings    = {
      "subject" = "subject"
    }
  }
}

```
### Expected Behavior
ClaimMappings/ListMappings should be configured properly

### Actual Behavior

```bash
 $ terraform plan

Error: Incorrect attribute value type

  on consul.tf line 13, in resource "consul_acl_auth_method" "jwt":
  13:   config = {
  14:     JWKSURL           = "https://somedomain.com/identity/oidc/.well-known/keys"
  15:     JWTSupportedAlgs = "RS256"
  16:     BoundIssuer       = "https://somedomain.com"
  17:     ClaimMappings     = {
  18:       "subject" = "subject"
  19:     }
  20:   }

Inappropriate value for attribute "config": element "ClaimMappings": string
required.
```

**Comment by remilapeyre** [comment]
Thanks for the feedback @pbusko! I checked with the Consul team and since the new auth methods have a more complex configuration that the kubernetes one, we won't be able to manage them using the current `config` attribute.

To fix this, #208 adds a new `config_json` attribute that will send it's data directly to Consul without validating them. This will make it possible to use an arbitrary complex configuration :)

**Comment by pbusko** [comment]
Thanks @remilapeyre!

Any estimate on when #208 will be released? We're heavily relying on this feature and currently have to use customised version of provider.
